pkg/proxy/service.go

```diff
@@ -214,19 +215,21 @@ type ServiceChangeTracker struct {
 	// items maps a service to its serviceChange.
 	items map[types.NamespacedName]*serviceChange
 	// makeServiceInfo allows proxier to inject customized information when processing service.
-	makeServiceInfo makeServicePortFunc
+	makeServiceInfo  makeServicePortFunc
+	serviceMapChange serviceMapChangeFunc
 	// isIPv6Mode indicates if change tracker is under IPv6/IPv4 mode. Nil means not applicable.
 	isIPv6Mode *bool
 	recorder   record.EventRecorder
 }
 
 // NewServiceChangeTracker initializes a ServiceChangeTracker
-func NewServiceChangeTracker(makeServiceInfo makeServicePortFunc, isIPv6Mode *bool, recorder record.EventRecorder) *ServiceChangeTracker {
+func NewServiceChangeTracker(makeServiceInfo makeServicePortFunc, isIPv6Mode *bool, recorder record.EventRecorder, serviceMapChange serviceMapChangeFunc) *ServiceChangeTracker {
 	return &ServiceChangeTracker{
-		items:           make(map[types.NamespacedName]*serviceChange),
-		makeServiceInfo: makeServiceInfo,
-		isIPv6Mode:      isIPv6Mode,
-		recorder:        recorder,
+		items:            make(map[types.NamespacedName]*serviceChange),
+		makeServiceInfo:  makeServiceInfo,
+		isIPv6Mode:       isIPv6Mode,
+		recorder:         recorder,
+		serviceMapChange: serviceMapChange,
```

Comment:
Maybe just a bit of a rename here to make it clear this is a function and not a change object.
```suggestion
		onServiceMapChange: onServiceMapChange,
```